src/placement/jump_map.c

```diff
@@ -338,45 +231,51 @@ set_used_targets(d_list_t *down_targets, struct pool_domain *selected_dom,
  */
 static void
 get_target(struct pool_domain *curr_dom, struct pool_target **target,
-	   uint64_t obj_key, uint8_t *dom_used, struct pl_obj_layout *layout,
-	   int shard_num)
+	   uint64_t obj_key, uint8_t *dom_used, uint8_t *tgts_used,
+	   struct pl_obj_layout *layout, int shard_num)
 {
+	int			range_set;
 	uint8_t                 found_target = 0;
-	uint8_t                 top = 0;
-	uint32_t                fail_num = 0;
 	uint32_t                selected_dom;
-	uint32_t                tgt_id;
 	struct pool_domain      *root_pos;
 
+	obj_key = crc(obj_key, shard_num);
 	root_pos = curr_dom;
 
 	do {
 		uint32_t        num_doms;
-		uint64_t        key;
 
 		/* Retrieve number of nodes in this domain */
 		if (curr_dom->do_children == NULL)
 			num_doms = curr_dom->do_target_nr;
 		else
 			num_doms = curr_dom->do_child_nr;
 
-		key = obj_key;
-		/* If choosing target in lowest fault domain level */
-
+		/* If choosing target (lowest fault domain level) */
 		if (curr_dom->do_children == NULL) {
-			uint32_t dom_id;
-			uint32_t        i;
+
+			uint32_t	fail_num = 0;
+			uint32_t	dom_id;
+			uint32_t	start_tgt;
+			uint32_t	end_tgt;
+
+			start_tgt = curr_dom->do_targets[0].ta_comp.co_id;
+			end_tgt = start_tgt + (num_doms-1);
+
+			range_set = isset_range(tgts_used, start_tgt, end_tgt);
+			if (range_set)
+				clrbit_range(tgts_used, start_tgt, end_tgt);
```

Comment:
does it mean we have colocated shards? If it is true then we should at least add a debug message for it.